projects/RabbitMQ.Client/client/impl/AsyncConsumerWorkService.cs

```diff
@@ -33,18 +40,30 @@ public Task Stop(IModel model)
         class WorkPool
         {
             readonly ConcurrentQueue<Work> _workQueue;
+            readonly ConcurrentBag<Work> _workBag;
             readonly CancellationTokenSource _tokenSource;
             readonly ModelBase _model;
             readonly CancellationTokenRegistration _tokenRegistration;
+            private readonly bool _runInParallel;
             volatile TaskCompletionSource<bool> _syncSource = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
             private Task _worker;
+            private readonly List<Task> _workTasks = new List<Task>();
```

Comment:
`_workTasks` can be made into a local.

Reply:
If you do that than you are making a tradeoff between the list growing and the allocation per iteration. The GC costs can quickly become higher. If the suggestion of limiting the concurrency that I made are taken into account the list can have a fixed size and never need to grow plus you don't create Gen0 garbage